feat: support disabling clock emergency sound

tom-anders opened the pull request:
Closes #785

Needs https://github.com/lichess-org/lila/pull/15613

tom-anders commented:
(fixed formatting)

veloce commented on lib/src/widgets/countdown_clock.dart:
```diff
@@ -55,6 +59,7 @@ const _emergencyDelay = Duration(seconds: 20);
 class _CountdownClockState extends ConsumerState<CountdownClock> {
   Timer? _timer;
   Duration timeLeft = Duration.zero;
+  bool _emergencySoundEnabled = true;
```
You don't need that local property since it doesn't change. Use directly `widget.emergencySoundEnabled` instead.

veloce reviewed (commented):
Thanks! Will wait for the deploy of server change before merging this.

tom-anders commented:
Thanks for the quick feedback! Addressed all your comments.

@veloce if you don't mind, I might have a look at https://github.com/lichess-org/mobile/issues/759 next, it seems to require similar changes (probably also a similar server change as well

veloce reviewed (approved):
lgtm
